### PR TITLE
logging: consistent error attribute when emitted through logr

### PR DESCRIPTION
**Pull request by giorio94**
Extend the log attributes mangling logic to ensure consistency in the attribute key used to identify an error. Indeed, the `logr.Error` method takes an error as first parameter, and the slog adapter converts it into an attribute identified as `err`. Let's replace that into `error` to ensure consistency with all the other occurrences.

### CODE REVIEW

File: pkg/logging/slog.go [hunk 1]
```diff
@@ -95,6 +100,11 @@ func replaceLevel(groups []string, a slog.Attr) slog.Attr {
 			Key:   a.Key,
 			Value: slog.StringValue(strings.ToLower(a.Value.String())),
 		}
+	case logrErrorKey:
```
Comment (mhofstetter) [inline pkg/logging/slog.go:103]: the name of this function (`replaceLevel`) is no longer correct (probably already before this change).

File: pkg/logging/slog.go [hunk 2]
```diff
@@ -110,6 +120,11 @@ func replaceLevelAndDropTime(groups []string, a slog.Attr) slog.Attr {
...
```
Comment (mhofstetter) [inline pkg/logging/slog.go]: the name of this function (`replaceLevelAndDropTime`) is no longer correct (probably already before this change).

**Review by mhofstetter — COMMENTED** [review]
Thanks! Only two questions regarding the naming of the "replace"-functions. Maybe it's worth to adjust them to something more generic? (`replaceFuncWithoutTimestamp` & `replaceFunc` ?)

File: pkg/logging/slog.go (hunk 1 above)
Comment (giorio94) [reply pkg/logging/slog.go:103]: Yeah, makes sense. I've generalized the two function names, and made one build on top of the other to ensure consistency and reduce code duplication.

**Review by mhofstetter — APPROVED** [review]
LGTM, thanks!

**Pull request merged**